contracts/extensions/FundingQueue.sol

```diff
@@ -283,25 +283,45 @@ contract FundingQueue is ColonyExtension, PatriciaTreeProofs {
 
     assert(proposal.totalPaid <= proposal.totalRequested);
 
-    if (proposal.totalPaid == proposal.totalRequested) {
-      proposal.state = ProposalState.Completed;
-
-      queue[HEAD] = queue[_id];
-      delete queue[_id];
-
-      //  May be the null proposal, but that's ok
-      proposals[queue[HEAD]].lastUpdated = updateTime;
+    try
```

Comment:
As per our discussion, we discussed not using `try/catch` and instead checking the permission explicitly:

```
if (!colony.hasUserRole(address(this), proposal.domainId, FUNDING_ROLE)) {
  // Cancel proposal
}
```

I think that'll be more succinct and readable.

Reply:
Hm, I was writing a big long answer saying how that's how I started, but doing this check requires six external calls, but while writing it I've realised you are right, this is enough. That domain does indeed have to be the domain we have an explicit permission in, and so we don't have to check the inheritance etc, which can be left to be done in the main call.

Reply:
Fab u lous